File: leetcode75/1768. Merge Strings Alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = ''.join(a + b for a, b in zip(word1, word2))
        return result + word1[len(word2):] + word2[len(word1):]

# or

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = []

        for a,b in zip(word1, word2):
            result.append(a)
            result.append(b)

        result.extend(word1[len(word2):])
        result.extend(word2[len(word1):])

        return "".join(result)

File: leetcode75/1768. Merge Strings Alternately/Solution.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        
        int length1 = word1.Length;
        int length2 = word2.Length;
        var result = new char[length1 + length2];
        int i = 0, j = 0, k = 0;

        while (i < length1 && j < length2) {
            result[k++] = word1[i++];
            result[k++] = word2[j++];
        }

        while(i < length1) result[k++] = word1[i++];
        while(j < length2) result[k++] = word2[j++];

        return new string(result);
    }
}
